### PR TITLE
feat: Adding x,y,width and height inputs to position components on Dev Tools

### PR DESCRIPTION
**Pull request by erickzanardo**
# Description

Adds fields to allow developers to change the x, y, width and height of position components from the dev tools.

https://github.com/user-attachments/assets/70f6fe15-488c-4eb2-83bf-6cdc050526d8

## Checklist

- [x] I have followed the [Contributor Guide] when preparing my PR.
- [x] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [x] I have updated/added relevant examples in `examples` or `docs`.

## Breaking Change?

- [ ] Yes, this PR is a breaking change.
- [x] No, this PR is not a breaking change.

## Related Issues

[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Conventional Commit]: https://conventionalcommits.org
[CHANGELOG]: https://github.com/flame-engine/flame/blob/main/CHANGELOG.md

### CODE REVIEW

File: packages/flame/lib/src/devtools/connectors/position_attributes_component_connector.dart [hunk 1]
```diff
@@ -0,0 +1,69 @@
+import 'dart:convert';
+import 'dart:developer';
+
+import 'package:flame/components.dart';
+import 'package:flame/src/devtools/dev_tools_connector.dart';
+
+class PositionAttributesComponentConnector extends DevToolsConnector {
```
Comment (spydon) [inline packages/flame/lib/src/devtools/connectors/position_attributes_component_connector.dart:7]:
```suggestion
class PositionComponentAttributesConnector extends DevToolsConnector {
```

File: packages/flame/lib/src/devtools/dev_tools_connector.dart [hunk 2]
```diff
@@ -29,4 +30,23 @@ abstract class DevToolsConnector {
 
   /// Here you can do clean-up before a new game is set in the connector.
   void disposeGame() {}
+
+  /// Finds a component in the game tree by its id.
+  ///
+  /// Returns the component if found, otherwise null.
+  T? findGameComponent<T extends Component>(int? id) {
```
Comment (spydon) [inline packages/flame/lib/src/devtools/dev_tools_connector.dart:37]:
```suggestion
  T? findComponent<T extends Component>(int? id) {
```

File: packages/flame_devtools/lib/repository.dart [hunk 3]
```diff
@@ -76,4 +76,56 @@ sealed class Repository {
...
+    final potentialPositionComponentResponse =
+        await serviceManager.callServiceExtensionOnMainIsolate(
+      'ext.flame_devtools.getPositionComponentAttributes',
+      args: {'id': id},
+    );
+
+    return PositionComponentAttributes.fromJson(
+      potentialPositionComponentResponse.json!,
+    );
+  }
```
Comment (spydon) [inline packages/flame_devtools/lib/repository.dart:92]:
```suggestion
  final positionComponentAttributesProvider = FutureProvider.family<PositionComponentAttributes, int>((ref, id) async {
      final potentialPositionComponentResponse =
          await serviceManager.callServiceExtensionOnMainIsolate(
        'ext.flame_devtools.getPositionComponentAttributes',
        args: {'id': id},
      );

      return PositionComponentAttributes.fromJson(
        potentialPositionComponentResponse.json!,
      );
    },
    )
  }
```

File: packages/flame_devtools/lib/widgets/position_component_attributes_form.dart [hunk 4]
```diff
@@ -0,0 +1,114 @@
+import 'package:flame_devtools/repository.dart';
+import 'package:flame_devtools/widgets/incremental_number_form_field.dart';
+import 'package:flutter/material.dart';
+
+class PositionComponentAttributesForm extends StatefulWidget {
```
Comment (spydon) [inline packages/flame_devtools/lib/widgets/position_component_attributes_form.dart:5]:
```suggestion
class PositionComponentAttributesForm extends ConsumerWidget {
```

File: packages/flame_devtools/lib/widgets/position_component_attributes_form.dart [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+      _attributesFuture = Repository.getPositionComponentAttributes(
+        id: widget.componentId,
+      );
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return FutureBuilder<PositionComponentAttributes>(
+      future: _attributesFuture,
```
Comment (spydon) [inline packages/flame_devtools/lib/widgets/position_component_attributes_form.dart:45]:
```suggestion
  @override
  Widget build(BuildContext context) {
    final positionComponentAttributesData = ref.watch(positionComponentAttributesProvider);
    return positionComponentAttributesData.when(
      data: (positionComponentAttributes) {
         ...
      },
      loading: ...
      error: ...
    );
  }
```

**Review by spydon — COMMENTED** [review]
There is no polling for the attributes if they change within the game currently right? Just check if I missed that, or if it's maybe planned as a follow-up? :)

**Comment by erickzanardo** [comment]
> There is no polling for the attributes if they change within the game currently right? Just check if I missed that, or if it's maybe planned as a follow-up? :)

I did not implemented that indeed, tbh, that didn't even crossed my mind hahha, I guess my creativity is not what it used to be...

It is a good idea though! Should we make a refresh button like the component tree or just go straight into a pooling???

**Comment by spydon** [comment]
> It is a good idea though! Should we make a refresh button like the component tree or just go straight into a pooling???

I think polling would be more user friendly, but maybe problematic when the user wants to set the values of something that is moving/resizing... Maybe just not updating the values while the user is in the text field.

**Comment by ufrshubham** [comment]
@erickzanardo will it be possible to add some titles to the properties? Something like this:
image

**Comment by erickzanardo** [comment]
> @erickzanardo will it be possible to add some titles to the properties? Like `Transform`, `Position`, `Size`, something like this: image

Changed to look like this:

Captura de Tela 2024-08-11 às 21 07 09

**Review by spydon — APPROVED** [review]
Lgtm!!

**Pull request merged**